libsql/src/sync.rs

```diff
@@ -256,7 +260,10 @@ impl SyncContext {
                 let frame = hyper::body::to_bytes(res.into_body())
                     .await
                     .map_err(SyncError::HttpBody)?;
-                return Ok(frame);
+                return Ok(Some(frame));
+            }
+            if res.status() == StatusCode::BAD_REQUEST {
+                return Ok(None);
             }
```

Comment:
I am not a big fan of catch all BAD_REQUEST here since there are other areas that can produce this code and we end up eating the response. Should the server return 404 if the frame doesn't exist? Or maybe it should return 200 + an error code.